desc/objectives/_stability.py

```diff
@@ -367,3 +369,211 @@ def compute(self, params, constants=None):
+        self._iota_keys = ["iota", "iota_r", "iota_rr"]  # might not need all of these
+        iota_profiles = get_profiles(self._iota_keys, obj=eq, grid=iota_grid)
+        iota_transforms = get_transforms(self._iota_keys, obj=eq, grid=iota_grid)
+
+        ## Separate grid to calculate the right length scale for normalization
+        len_grid = LinearGrid(rho=1.0, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+        self._len_keys = ["a"]
+        len_profiles = get_profiles(self._len_keys, obj=eq, grid=len_grid)
+        len_transforms = get_transforms(self._len_keys, obj=eq, grid=len_grid)
+
+        # make a set of nodes along a single fieldline
+        zeta = np.linspace(-self.zetamax, self.zetamax, self.nzeta)
+        rho, alpha, zeta = np.reshape(
+            np.broadcast_arrays(self.rho, self.alpha, zeta), (3, -1)
+        )
+        fieldline_nodes = np.array([rho, alpha, zeta]).T
+
+        self._dim_f = 1
+        self._data_keys = ["ideal_ball_gamma2"]
+
+        self._args = get_params(
+            self._iota_keys + self._len_keys + self._data_keys,
+            obj="desc.equilibrium.equilibrium.Equilibrium",
+            has_axis=False,
+        )
+
+        self._constants = {
+            "iota_transforms": iota_transforms,
+            "iota_profiles": iota_profiles,
+            "len_transforms": len_transforms,
+            "len_profiles": len_profiles,
+            "fieldline_nodes": fieldline_nodes,
+            "quad_weights": 1.0,
+        }
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """
+        Compute the ballooning stability growth rate.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of equilibrium degrees of freedom, eg Equilibrium.params_dict
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self.constants
+
+        Returns
+        -------
+        gamma : ndarray
+            ideal ballooning growth rate.
+
+        """
+        from desc.equilibrium.coords import compute_theta_coords
+
+        eq = self.things[0]
+
+        if constants is None:
+            constants = self.constants
+        # we first compute iota on a uniform grid to get correct averaging etc.
+        iota_data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._iota_keys,
+            params=params,
+            transforms=constants["iota_transforms"],
+            profiles=constants["iota_profiles"],
+        )
+
+        len_data = compute_fun(
+            "desc.equilibrium.equilibrium.Equilibrium",
+            self._len_keys,
+            params=params,
+            transforms=constants["len_transforms"],
+            profiles=constants["len_profiles"],
+        )
+
+        # Now we compute theta_DESC for given theta_PEST
+        iota = iota_data["iota"][0]
```

Comment:
this assumes you only have a single flux surface, what if there are multiple rho values?

Reply:
You need a separate objective for each rho. This is what we decided at Simons Australia. Another reason to generalize this logic in #1024 after this branch is merged.

Reply:
building transforms one flux surface at a time is slower than building one for the full grid. it would be a speed improvement to use `get_rtz_grid` in `desc.equilibrium.coords`. You can specify the kwarg `iota=data["iota"]` to make it faster as well if you already have iota evaluated on the full grid.